sponsorship_switzerland/models/icp_lifecycle.py

```diff
@@ -0,0 +1,38 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2016 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Philippe Heer <heerphilippe@msn.com>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+from openerp import models
+
+
+class ProjectLifecycle(models.Model):
+    """ Send Communication when icp lifecycle is received. """
+    _inherit = 'compassion.project.ile'
+
+    def process_commkit(self, commkit_data):
+        ids = super(ProjectLifecycle, self).process_commkit(commkit_data)
+
+        lifecycle_event_type = commkit_data.get('ICPLifecycleEventType')
```

Comment:
Instead of looking for the information instead the the commkit data you could use
`for lifecycle in self.browse(ids):`
And use all information of the lifecycle record. (lifecycle.type, lifecycle.icp_id, etc..)